### ISSUE TITLE
Adding target language token in mBART model.

### ISSUE THREAD

**Issue by asking28** [huggingface/transformers#13741]
## Environment info

- `transformers` version: 4.10.3
- Platform: Google colab
- Python version: 3.9
- PyTorch version (GPU?):  1.9.0+cu102
- Tensorflow version (GPU?):
- Using GPU in script?: No
- Using distributed or parallel set-up in script?:

### Who can help

@patrickvonplaten @patil-suraj @sgugger 

## Information

Model I am using : mBART

The problem arises when using:
*  the official example scripts: (give details below)

`>>> from transformers import MBartTokenizer
>>> tokenizer = MBartTokenizer.from_pretrained('facebook/mbart-large-en-ro', src_lang="en_XX", tgt_lang="ro_RO")
>>> example_english_phrase = " UN Chief Says There Is No Military Solution in Syria"
>>> expected_translation_romanian = "Şeful ONU declară că nu există o soluţie militară în Siria"
>>> inputs = tokenizer(example_english_phrase, return_tensors="pt)
>>> with tokenizer.as_target_tokenizer():
...     labels = tokenizer(expected_translation_romanian, return_tensors="pt")
>>> inputs["labels"] = labels["input_ids"]`

## To reproduce

1. Documentation mentions "The tokenization method is <tokens> <eos> <language code> for source language documents, and <language code> <tokens> <eos>` for target language documents." [https://huggingface.co/transformers/model_doc/mbart.html#transformers.MBartTokenizer.as_target_tokenizer](url)
2. Were as the source code adds only suffix when "as_target_tokenizer" is called to tokenize target sentence in the format:
`    @contextmanager
    def as_target_tokenizer(self):
        """
        Temporarily sets the tokenizer for encoding the targets. Useful for tokenizer associated to
        sequence-to-sequence models that need a slightly different processing for the labels.
        """
        self.set_tgt_lang_special_tokens(self.tgt_lang)
        yield
        self.set_src_lang_special_tokens(self.src_lang)

    def set_src_lang_special_tokens(self, src_lang) -> None:
        """Reset the special tokens to the source lang setting. No prefix and suffix=[eos, src_lang_code]."""
        self.cur_lang_code = self.lang_code_to_id[src_lang]
        self.prefix_tokens = []
        self.suffix_tokens = [self.eos_token_id, self.cur_lang_code]

    def set_tgt_lang_special_tokens(self, lang: str) -> None:
        """Reset the special tokens to the target language setting. No prefix and suffix=[eos, tgt_lang_code]."""
        self.cur_lang_code = self.lang_code_to_id[lang]
        self.prefix_tokens = []
        self.suffix_tokens = [self.eos_token_id, self.cur_lang_code]`
3. It should add a target language code as prefix and eos token as suffix. [https://huggingface.co/transformers/_modules/transformers/models/mbart/tokenization_mbart.html#MBartTokenizer.as_target_tokenizer](url)

## Expected behavior
It should add a target language code as prefix and eos token as suffix as per documentation.
